File: src/WebApp/src/app/Models/UserDto.ts
export class UserDto{

    constructor(
        private userId : number,
        private username : string,
        private email : string,
        private role : string,
        private isActive : boolean) {}


        getUserId() : number{
            return this.userId;
        }

        getUsername() : string {
            return this.username;
        }

        getEmail() : string {
            return this.email;
        }

        getRole() : string {
            return this.role;
        }

        userActive() : boolean{
            return this.isActive;
        }
}

File: src/Backend/Backend/Models/Dto/UserDto.cs
using System.Text.Json.Serialization;

namespace Backend.Models.Dto
{
    public class UserDto
    {
        public int UserId { get; set; }
        public string Username { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public UserRole Role { get; set; }
        public bool IsActive { get; set; }
    }
}
